src/doc.d

```diff
@@ -2479,6 +2499,73 @@ extern (C++) void highlightCode(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 i = j - 1;
             }
         }
+        else if (!resolvedTemplateParameters)
+        {
+            // hunt for template declarations:
+            for (size_t symi = 0; symi < a.dim; symi++)
```

Comment:
It would be awesome of the C++ array objects offered an opSlice or something so we could do a plain foreach on them. Or even range primitives!

A lot of the compiler modifiers arrays as you iterate and there you want a traditional for loop, but these other ones could be pretty easily Dified a bit.